apps/frontend/src/libs/modules/map-client/map-client.module.ts

```diff
@@ -0,0 +1,322 @@
+import mapboxgl from "mapbox-gl";
+
+import { config } from "~/libs/modules/config/config.js";
+
+import {
+	CONTROL_IDS,
+	DEFAULT_GEOLOCATE_CONTROL_OPTIONS,
+	DEFAULT_MAP_OPTIONS,
+	DEFAULT_MARKER_OPTIONS,
+	DEFAULT_NAVIGATION_CONTROL_OPTIONS,
+	DEFAULT_SCALE_CONTROL_OPTIONS,
+	GEOLOCATE_TRIGGER_DELAY,
+} from "./libs/constants/constants.js";
+import {
+	type Constructor,
+	type ControlPosition,
+	type GeolocateControlOptions,
+	type IControl,
+	type MapBoxGLWithToken,
+	type MapOptions,
+	type Marker,
+	type MarkerOptions,
+	type NavigationControlOptions,
+	type PointGeometry,
+	type ScaleControlOptions,
+} from "./libs/types/types.js";
+
+import "mapbox-gl/dist/mapbox-gl.css";
+
+class MapClient {
+	private controls = new Map<string, IControl>();
+	private defaultGeolocateControlOptions: GeolocateControlOptions & {
+		autoTrigger?: boolean;
+		position?: ControlPosition;
+	};
+	private defaultMapOptions: Omit<MapOptions, "container">;
+	private defaultMarkerOptions: MarkerOptions;
+	private defaultNavigationControlOptions: NavigationControlOptions & {
+		position?: ControlPosition;
+	};
+	private defaultScaleControlOptions: ScaleControlOptions & {
+		position?: ControlPosition;
+	};
+	private map: mapboxgl.Map | null = null;
+	private markerClickHandler?: (
+		id: string,
+		coordinates: PointGeometry["coordinates"],
+	) => void;
+	private markers = new Map<string, Marker>();
+	private resizeObserver: null | ResizeObserver = null;
+	private shouldCenterOnUser: boolean = false;
+
+	public constructor({ token }: Constructor = {}) {
+		(mapboxgl as MapBoxGLWithToken).accessToken =
+			token || config.ENV.MAPBOX.ACCESS_TOKEN;
```

Comment:
Not sure why we need typecast here, shouldn't Mapbox already have accessToken in types. Also, we don't need many ways to pass the token, only one
```suggestion
	public constructor() {
		mapboxgl.accessToken = config.ENV.MAPBOX.ACCESS_TOKEN;
```

Reply:
About that...
@types/mapbox is now integrated into the mapbox-gl package itself. However, the core library doesn't explicitly expose the accessToken type. Instead, it's included as part of the MapOptions type — so I extracted and reused it from there to maintain correct typing.
<img width="837" height="104" alt="Знімок екрана 2025-08-09 223201" src="https://github.com/user-attachments/assets/8287d6b7-0ad7-4894-b363-d1b5687a5067" />

Reply:
oh, i see, ok then let's leave it as it is